edb/server/server.py

```diff
@@ -199,15 +205,30 @@ def on_binary_client_connected(self) -> str:
 
     def on_binary_client_authed(self):
         self._binary_proto_num_connections += 1
+
+        if self._auto_shutdown_handler:
```

Comment:
I'd put this into `on_binary_client_connected` instead, otherwise a delay in auth might cause premature termination.

Reply:
And I think the `_binary_proto_num_connections` should be moved there also.

Reply:
changed

Reply:
moving `_binary_proto_num_connections` caused some test to hang so I moved it back